src/App.js

```diff
@@ -1,26 +1,97 @@
-import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import React, { Component } from 'react';
+import 'bulma/css/bulma.css';
+import foodList from './foods.json';
+import FoodBox from './Components/FoodBox';
+import FoodForm from './Components/FoodForm';
+import FoodSearch from './Components/FoodSearch';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends Component {
+  state = {
+    foodState: foodList,
+    form: false,
+    filteredFoodListState: [],
+    searching: false,
+    todaysFoodList: [],
+  };
+
+  handleRenderForm = () => {
+    const stateCopy = { ...this.state };
+    stateCopy.form = !this.state.form;
+    this.setState(stateCopy);
+  };
+
+  handleAddNewFood = (newFoodItem) => {
+    const stateCopy = { ...this.state };
+    stateCopy.foodState = [...stateCopy.foodState, newFoodItem];
+    stateCopy.form = !stateCopy.form;
+    this.setState(stateCopy);
+  };
+
+  handleFilterFoods = (searchInput) => {
+    const stateCopy = { ...this.state };
+    const filteredFoodList = stateCopy.foodState.filter((foodItem) =>
+      foodItem.name.toLowerCase().includes(searchInput.toLowerCase())
+    );
+
+    stateCopy.filteredFoodListState = filteredFoodList;
+    stateCopy.searching = true;
+
+    this.setState(stateCopy);
+  };
+
+  handleAddFoodToTodaysList = (foodObject) => {
+    const stateCopy = { ...this.state };
+    foodObject.calories *= foodObject.quantity;
+    stateCopy.todaysFoodList.push(foodObject);
```

Comment:
One question to consider here is whether you would like to have instances of the same item listed separately, or whether you would like to increment the calories and quantity values of the item that already exists in the list, in case the item you are pushing already exists in the todaysFoodList.

You could do something like:

```
let foundFood = stateCopy.todaysFoodList.find(el => el.name === foodObject.name);

foodObject.calories *= food.quantity;

if (foundFood) {
    foundFood.quantity += food.quantity;
    foundFood.calories += food.calories;
} else {
    today.push(food);
}

this.setState({
    today: stateCopy.todaysFoodList
});
```